docs/source/considerations.rst

```diff
@@ -93,17 +93,38 @@ In fact, the only difference is the binding from Alice, in which we are using
 a lambda function for the handler.
 
 
+.. _shutting_down:
+
 Shutting down
 =============
 
-Although not covered in the examples until now (because many times you just
-want the multi-agent system to run forever until, perhaps, an event occurs),
-it is possible to actively kill the system using proxies:
+Sometimes, we will want to manually shut down a single agent or even a group
+of agents. In this section, we will see how to do it.
+
+Shutting down agents
+--------------------
+
+If we want to end the execution of a specific agent in our system, we can do
+it by calling the `agent.shutdown` method:
 
 .. literalinclude:: ../../examples/shutdown.py
 
-.. note:: Shutting down the nameserver will result in all agents registered in
-   the name server being shut down as well.
+Shutting down name servers
+--------------------------
+
+We learned earlier that name servers run just like any other agent in our
+system, even though with specific functionality.
+
+Shutting down a name server will result in all agents registered in that name
+server being shut down as well. This allows us to easily shutdown groups of
+agents at the same time, by registering them in different name servers when
+created.
+
+.. literalinclude:: ../../examples/various_nameservers.py
+
+.. note:: As seen in the example above, that two agents are registered on
+   different name servers does not mean that we cannot stablish connections
+   between them.
```

Comment:
Mmmmhh... I can't think of a better example than the one I currently have. However, I think it could be useful (specially for "learning by example"). I've added some docstrings stating that the example is "relatively complex". However, if you feel the example and some explanations of what we are doing on it. In any case, if you feel the example should not be added to the project I've no problem taking it out.